### ISSUE TITLE
Detect and handle `docker compose` setups (grouping)

### ISSUE THREAD

**Issue by bittner** [RedSoftwareSystems/easy_docker_containers#42]
I like the way the menu handles context actions for a container:

1. You click on the task bar icon, the list of containers appears.
2. You click on a container, the container visually expands from the bottom, and a list of options/actions appears.

That's a great UX for independent containers!

## Accommodate Docker Compose with the same idea

I believe, the same idea could be used to accommodate groups of containers that are defined by Docker Compose. Those groups are listed by the `docker compose ls` command.

IIUC, the group is already detected, which results in displaying `<container group> ℴ <container name>` menu items.

- I would find it helpful if containers of the same "container group" were grouped together, expandable/collapsible as a whole (list of containers).
- In addition, there could be commands that apply to the entire Docker Compose setup in that intermediate menu.

**Comment by Ranger-NF** [comment]
Yeah, I too wish this was a feature. I use (Immich)[https://github.com/immich-app/immich] using docker compose. Having an easy to handle such container would be so helpful.

**Comment by kiuma** [comment]
Hello I already wanted to develop the docker compose integration.
Currently I'm in the process of completing a plugin refactoring for Strapi
about authorization (Strapi is a CMS/data appcelerator).

In a few days I'll start to work on plugin enhancement, I'll also add
start/stop all containers that can be done only after compose support.

Stay tuned,
Andrea

Il sab 27 gen 2024, 14:39 Ranger ***@***.***> ha scritto:
